### ISSUE TITLE
[Bug]: The terraform-provider-aws_v5.4.0_x5 plugin crashed

### ISSUE THREAD

**Issue by tasogare0919** [hashicorp/terraform-provider-aws#32212]
### Terraform Core Version

1.3.5

### AWS Provider Version

5.4.0

### Affected Resource(s)

* aws_appautoscaling_target
* aws_lambda_permission
* aws_iam_policy
* aws_iam_user_policy_attachment
* aws_iam_role_policy_attachment
* aws_iam_role

### Expected Behavior

That the terraform plan can be executed because nothing has been changed in the configuration.

### Actual Behavior

I confirmed that terraform plan fails with the following message.

```sh
Error: The terraform-provider-aws_v5.4.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Relevant Error/Panic Output Snippet

```shell
goroutine 7169 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless.(*resourceCollection).Read(0x0?, {0xf5cfa40, 0xc00a14b5f0}, {{{{0xf5df930, 0xc00a14a900}, {0xc2c7ca0, 0xc00a13fd40}}, {0xf5e96c8, 0xc0047858b0}}, 0xc0077619f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/service/opensearchserverless/collection.go:208 +0x2ad
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read.func1({0xf5cfa40?, 0xc00a14b5f0?}, {{{{0xf5df930, 0xc00a14a900}, {0xc2c7ca0, 0xc00a13fd40}}, {0xf5e96c8, 0xc0047858b0}}, 0xc0077619f0, {{{0x0, ...}, ...}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:202 +0x89
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.interceptedHandler[...].func1({{{{0xf5df930, 0xc00a14a900}, {0xc2c7ca0, 0xc00a13fd40}}, {0xf5e96c8, 0xc0047858b0}}, 0xc0077619f0, {{{0x0, 0x0}, {0x0, ...}}, ...}}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:105 +0x2f6
github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider.(*wrappedResource).Read(0xc00a126400, {0xf5cfa40, 0xc00a11b350}, {{{{0xf5df930, 0xc00a14a900}, {0xc2c7ca0, 0xc00a13fd40}}, {0xf5e96c8, 0xc0047858b0}}, 0xc0077619f0, ...}, ...)
	github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider/intercept.go:206 +0x157
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0040cc000, {0xf5cfa40, 0xc00a11b350}, 0xc00a11b440, 0xc00a16b3f8)
	github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/fwserver/server_readresource.go:101 +0x617
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0xc0040cc000, {0xf5cfa40?, 0xc00a11b200?}, 0xc00a1261c0)
	github.com/hashicorp/terraform-plugin-framework@v1.3.1/internal/proto5server/server_readresource.go:56 +0x27b
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ReadResource({0xc005a1c1e0, 0xc005a1c240, {0xc001f677c0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5muxserver/mux_server_ReadResource.go:26 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0040f4960, {0xf5cfa40?, 0xc00a11a720?}, 0xc009ed9f80)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0xdd184e0?, 0xc0040f4960}, {0xf5cfa40, 0xc00a11a720}, 0xc003e43030, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:383 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc008a31[680](https://github.com/xxx/xxx/actions/runs/5352528884/jobs/9749903331#step:6:681), {0xf5e0640, 0xc008a58340}, 0xc00a11e6c0, 0xc006b81b90, 0x163b5ed0, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf0
google.golang.org/grpc.(*Server).handleStream(0xc008a31680, {0xf5e0640, 0xc008a58340}, 0xc00a11e6c0, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a

Error: The terraform-provider-aws_v5.4.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
`
```

### Terraform Configuration Files

After running terraform apply with the following OpenSearch Serverless Collection settings, terraform provider crashes.

```hcl
resource "aws_opensearchserverless_collection" "example" {
  name = "example"
  depends_on = [aws_opensearchserverless_security_policy.example]
}
resource "aws_opensearchserverless_security_policy" "example" {
  name = "example"
  type = "encryption"
  policy = jsonencode({
    Rules = [
      {
        Resource = [
          "collection/example"
        ],
        ResourceType = "collection"
      }
    ],
    AWSOwnedKey = false
    KmsArn = [KMS ARN]
  })
}
```

### Steps to Reproduce

The following work was done in phases.

1, Create data access policy
2, Create network policy and encryption policy
3, Set depends_on in collection definitions

After running terraform apply above, it started crashing.

### Would you like to implement a fix?

None

**Comment by trevorrea** [comment]
Could be linked to https://github.com/hashicorp/terraform-provider-aws/issues/31722

If you are limiting memory on your Terraform runner increase it and see if the error still occurs.

**Comment by justinretzolk** [comment]
Hey @tasogare0919 👋 Thank you for taking the time to raise this! So that we have the necessary information in order to look into this, can you supply debug logs (redacted as needed) as well?

**Comment by tasogare0919** [comment]
Hi, @justinretzolk 
Thank you for your reply!
I was checking the contents of the debug log of the terraform plan and found a message in the log about lack of permissions for OpenSearch Serverless, which crashed immediately after that.
Therefore, I granted OpenSearch Serverless read permission to IAM for terraform plan, and the crash was resolved.
I am also able to run terraform apply.
Therefore, may I close this issue?
